### ISSUE TITLE
Communication channel machine provider / customer needed

### ISSUE THREAD

**Issue by HansHagberg** [DeepBrainChain/deepbrainchain-release#14]
There is need for manual communication between machine providers (node owners) and the customers putting the tasks on the system.  Issue #12 highlights one of the reasons for this. (buggy task with poor performance that also crashed the node)

The tasks are often custom software written by the customers. AI computing is a research area where a lot of development and experimentation is done. The software **will** contain **bugs or undesired behavior**. Need for **tuning and improvement in performance is a normal situation**.  This will put a distributed system at a disadvantage compared to owned hardware or rented (AWS). The buyers can't monitor the systems in real time like a direct accessed system.

By introducing some feature that allows human communication between the buyer and the node owner, this can be turned into an advantage instead. Some node owners will be interested in analyzing and monitoring the running tasks in order to improve the system. Some node owners will be technically savvy and willing to provide support and help the client with advice and troubleshooting. 

In the longer perspective, this can be made an integrated feature of the system. Having an integrated way to get active human support is a feature not available elsewhere and it can be worth a lot to a customer with problems. If the dashboard supports tagging of specific nodes as offering support, then the buyers will be able to select systems where he can communicate with the machine provider. In addition to just tagging the nodes with "support", there should also be room for a free text field what kind of support is possible (_e.g. Linux sysadm, performance analysis, AI training, etc._). Another field that lists language skills is also good to have (_the customer may need chinese support_).

**Requirements for the communication feature**
----------------------------------------------
- Mail.  This entire idea can be tested first by just connecting machine provider / buyer through an automatic mail when a task is started. Both parties will get each others mail addresses and can communicate and send files through mail and any other channel they agree on. The KYC data already has the needed contact info. Very quick to implement.

- Machine providers need to be able to tag their nodes if they are interested in giving manual support to the task issuer.  **Attributes**: support on/off,  languages,  type of support (free text)

- Feature to send files from the node to the outside world. Making this one-way will increase security.

- Simple text message real time chat is very useful. Can probably be handled through some existing external application. Security may be an issue

- Recurring tasks.  A buyer in need of support will probably want to send many short tasks over a period of time as a part of a typical development phase. He will want to always use the same node and not an automatically selected node. There must be a way to make it easy to only use the same specific node out of the thousands that will eventually be on the network. The normal automatic selection will not work.

- Economic incentive.  It can be debated how to give incentive to node owners to provide qualified manual support. Initially, when there is a shortage of tasks, special mechanisms may not be needed. The support option will be a way for node owners to attract tasks they would otherwise not get. If manual support becomes popular and is used a lot, it will be unsustainable for the node owners to work for free for hours. Just raising the normal node price/h will not work because then the node will never get the ordinary tasks without support. In this situation, a separate price mechanism for support (if used ) may be needed. In the long perspective (when the dbc network is huge), the manual support feature can turn into a killer app of it's own that creates a new profession (distributed manual AI computing support). 
In the short perspective, the project may want to assign personnel for this as a way to attract initial customers. As issue #12 has shown, there exists problems already.

**Comment by brightzhang2018** [comment]
@HansHagberg , yes . the communication btw AI user and AIM provider is very important  and high effective! this requirement we have already considered and make it in the future version.

**Comment by HansHagberg** [comment]
Auto translated text:

机器供应商（节点所有者）和客户之间需要手动通信将任务放在系统上。 问题＃12强调了其中的一个原因。 （性能不佳的错误任务也会导致节点崩溃)

这些任务通常是由客户编写的自定义软件。 人工智能计算是一个研究领域，进行了大量的开发和实验。 该软件将包含错误或不受欢迎的行为。 需要调整和改善性能是一个正常的情况。 与拥有的硬件或租用的(AWS)相比，这会使分布式系统处于劣势。 买家不能像直接访问系统一样实时监控系统。

通过引入一些允许买方和节点所有者之间的人际沟通的功能，这可以变成一个优势，而不是。 一些节点所有者会对分析和监控运行任务感兴趣以改进系统。 一些节点所有者将在技术上精通，并愿意提供支持，并帮助客户端提供建议和故障排除。

在较长的角度来看，这可以使系统的集成功能。 有一个集成的方式来获得积极的人的支持是一个功能无法在其他地方，它可以是值得有问题的客户很多。 如果仪表板支持将特定节点标记为提供支持，那么买家将能够选择可以与机器提供商通信的系统。 除了只用"支持"标记节点之外，还应该有空间为自由文本字段提供什么样的支持（例如Linux sysadm，性能分析，AI训练等）。）。 另一个列出语言技能的领域也很好（客户可能需要中文支持）。

通信功能的要求

邮件 
这整个想法可以首先被测试通过连接机器提供者或买家通过一个自动邮件，当任务开始时。 双方将获得对方的邮件地址，并可以通过邮件和他们同意的任何其他渠道沟通和发送文件。 KYC数据已经有了所需的联系信息。 非常快的实现。

机器提供程序需要能够标记其节点，如果他们有兴趣向任务颁发者提供手动支持。 属性：支持开/关，语言，支持类型（自由文本)

功能将文件从节点发送到外部世界。 这样做会增加安全性。

简单的短信/实时聊天是非常有用的。 可能可以通过一些现有的外部应用程序来处理。 安全可能是一个问题

重复任务。 
需要支持的买家可能会希望在一段时间内发送许多短任务，作为典型开发阶段的一部分。 他会希望始终使用相同的节点，而不是自动选择的节点。 必须有一种方法可以很容易地只使用相同的特定节点，最终将在网络上。 正常的自动选择将不起作用。

经济激励。 
可以讨论如何激励节点所有者提供合格的手动支持。 最初，当任务短缺时，可能不需要特殊的机制。 支持选项将是节点所有者吸引他们否则不会得到的任务的一种方式。 如果手动支持变得流行，并使用了很多，这将是不可持续的节点业主免费工作时间。 只是提高普通节点的价格/h不会工作，因为那么节点将永远不会得到普通的任务没有支持。 在这种情况下，可能需要一个单独的价格机制来支持（如果使用）。 在长期的角度来看（当dbc网络是巨大的），手动支持功能可以变成它自己的一个杀手级的应用程序，创建一个新的职业（分布式手动AI计算支持）。
在短期的角度来看，该项目可能需要为此分配人员，以吸引最初的客户。 正如问题#12所示，已经存在问题。
